src/main/java/org/influxdb/dto/BoundParameterQuery.java

```diff
@@ -0,0 +1,125 @@
+package org.influxdb.dto;
+
+import com.squareup.moshi.JsonWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import okio.Buffer;
+
+public final class BoundParameterQuery extends Query {
+
+  private final Map<String, Object> params = new HashMap<>();
+
+  private BoundParameterQuery(final String command, final String database) {
+    super(command, database, true);
+  }
+
+  public String getParameterJsonWithUrlEncoded() {
+    try {
+      List<String> placeholders = parsePlaceHolders(getCommand());
```

Comment:
`parsePlaceHolders` is only used to create a list for `assurePlaceholdersAreBound`.
Could you please move `parsePlaceHolders` to be called from inside `assurePlaceholdersAreBound`?

Reply:
Relevant code has been removed